sdk/src/test/scala/com/horizen/account/utils/AccountMockDataHelper.scala

```diff
@@ -0,0 +1,38 @@
+package com.horizen.account.utils
+
+import com.horizen.account.FeeUtils
+import com.horizen.account.block.{AccountBlock, AccountBlockHeader}
+import com.horizen.account.history.AccountHistory
+import com.horizen.params.NetworkParams
+import org.mockito.Mockito
+import org.scalatestplus.mockito.MockitoSugar.mock
+import org.web3j.utils.Numeric
+import scorex.util.bytesToId
+
+import java.math.BigInteger
+import java.util.Optional
+
+case class AccountMockDataHelper(genesis: Boolean) {
+
+  def getMockedAccountHistory(block: Optional[AccountBlock]): AccountHistory = {
+    val history: AccountHistory = mock[AccountHistory]
+    Mockito.when(history.params).thenReturn(mock[NetworkParams])
+    if (genesis) {
+      Mockito.when(history.params.sidechainGenesisBlockParentId).thenReturn(bytesToId(new Array[Byte](32)))
+    }
+    Mockito.when(history.getBlockById(block.get.id)).thenReturn(block)
+    history
+  }
+
+  def getMockedBlock(baseFee: BigInteger, gasUsed: Long, blockId: scorex.util.ModifierId, parentBlockId: scorex.util.ModifierId): AccountBlock = {
+    val block: AccountBlock = mock[AccountBlock]
+    Mockito.when(block.header).thenReturn(mock[AccountBlockHeader])
+    Mockito.when(block.header.parentId).thenReturn(parentBlockId)
+    Mockito.when(block.id).thenReturn(blockId)
+    Mockito.when(block.header.baseFee).thenReturn(baseFee)
+    Mockito.when(block.header.gasUsed).thenReturn(gasUsed)
+    Mockito.when(block.header.gasLimit).thenReturn(FeeUtils.GAS_LIMIT)
```

Comment:
looks good.
As was agree offline let's add a couple of test case with the modified `header.gasLimit`.
To be sure that algorithm works as expected in case the `gasLimit` may change as well.